lib/ui/page/home/page/chat/widget/AudioPlayer.dart

```diff
@@ -0,0 +1,239 @@
+  final String? friendly_name;
+
+  String get path_or_url {
+    return path != null ? path! : url!;
+  }
+
+  @override
+  State<AudioPlayer> createState() => _AudioPlayerState();
+
+}
+
+class _AudioPlayerState extends State<AudioPlayer> {
+
+  /// Indicator whether this [PlayableAsset] is hovered.
+  bool _hovered = false;
+
+  /// [StreamSubscription] for the audio playback.
+  StreamSubscription? _audio;
+
+  /// rewind / forward slider limits
+  var _currentSliderValueMs = 0.0.obs;
+  var _maxSliderValueMs = 1000.0.obs;
+
+  /// current audio stream player
+  PlayerController? _audioStream;
+
+  @override
+  Widget build(BuildContext context) {
+    final style = Theme.of(context).style;
+
+    return Column(children: [
+      Row(children: [
+        MouseRegion(
+          onEnter: (_) => setState(() => _hovered = true),
+          onExit: (_) => setState(() => _hovered = false),
+          child: WidgetButton(
+            onPressed: _audio != null ? _stopAudio : _playAudio,
+            child: AnimatedContainer(
+              duration: const Duration(milliseconds: 200),
+              height: 48,
+              width: 48,
+              decoration: BoxDecoration(
+                shape: BoxShape.circle,
+                color:
+                    _hovered ? style.colors.backgroundAuxiliaryLighter : null,
+                border: Border.all(
+                  width: 2,
+                  color: style.colors.primary,
+                ),
+              ),
+              child: Center(
+                child: Icon(
+                  _audio != null
+                      ? Icons.pause_rounded
+                      : Icons.play_arrow_rounded,
+                  size: 36,
+                  color: const Color(0xFF1F3C5D),
+                ),
+              ),
+            ),
+          ),
+        ),
+        const SizedBox(width: 16),
+        SizedBox(
+          width: 300,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(
+                widget.friendly_name ?? "<no name>",
+                style: style.fonts.medium.regular.onBackground,
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis,
+              ),
+              const SizedBox(height: 4),
+              SelectionContainer.disabled(
+                child: StyledCupertinoButton(
+                  padding: EdgeInsets.zero,
+                  label: 'Download',
+                  style: style.fonts.smaller.regular.primary,
+                  onPressed: () async {
+                    var bn = widget.friendly_name ?? "audio.mp3";
+                    final file = await PlatformUtils.saveTo(
+                      '${Config.downloads}/${bn}',
+                    );
+                    if (file != null) {
+                      MessagePopup.success(
+                          '${widget.path_or_url} downloaded to ${Config.downloads}/${bn}');
```

Comment:
```suggestion
                      MessagePopup.success(
                          '${widget.path_or_url} downloaded to ${Config.downloads}/${bn}',);
```

https://dart.dev/effective-dart/style#consider-changing-your-code-to-make-it-more-formatter-friendly